File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumAceiteMotorista.js
var EnumAceiteMotoristaHelper = function () {
    this.Todos = "";
    this.Enviado = 0;
    this.Aceite = 1;
    this.Recusou = 2;
    this.Expirou = 3;
};

EnumAceiteMotoristaHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Escala enviada", value: this.Enviado },
            { text: "Motorista confirmou escala", value: this.Aceite },
            { text: "Motorista recusou escala", value: this.Recusou },
            { text: "Expirou tempo confirmação escala", value: this.Expirou },
        ];
    },

    obterOpcoesPesquisa: function () {
        return [{ text: "Todos", value: this.Todos }].concat(this.obterOpcoes());
    }
};

var EnumAceiteMotorista = Object.freeze(new EnumAceiteMotoristaHelper);

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumAceiteMotorista.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumAceiteMotorista
    {
        Enviado = 0,
        Aceite = 1,
        Recusou = 2,
        Expirou = 3
    }

    public static class EnumAceiteMotoristaHelper
    {
        public static string ObterDescricao(this EnumAceiteMotorista aceite)
        {
            switch (aceite)
            {
                case EnumAceiteMotorista.Enviado: return "Escala enviada";
                case EnumAceiteMotorista.Aceite: return "Motorista confirmou escala";
                case EnumAceiteMotorista.Recusou: return "Motorista recusou escala";
                case EnumAceiteMotorista.Expirou: return "Expirou tempo confirmação escala";
                default: return string.Empty;
            }
        }
    }
}
